src/core/Menu/Menu.tsx

```diff
@@ -91,7 +109,35 @@ export const Menu = (props: MenuProps) => {
       ref={menuRef}
       {...rest}
     >
-      {children}
+      {React.Children.map(children, (item) => {
+        if (React.isValidElement(item)) {
```

Comment:
It is recommended to fail fast as it allows to avoid deep if conditions. Here better write `if (!React.isValidElement(item)) return item;`